elastic/kibana
# [RAM][Maintenance Window][Test PR] Maintenance Window Scoped Query Task Manager Changes

## Summary
Please only use this PR to test this feature, since this PR has both the frontend and backend changes. 

To review, use this PR instead: https://github.com/elastic/kibana/pull/172252

### To test:

ensure `IS_SCOPED_QUERY_ENABLED` in `x-pack/plugins/alerting/public/pages/maintenance_windows/constants.ts` is set to `true`

1. Make sure you are a platinum user
2. Create a maintenance window with a scoped query (I use `kibana.alerting.rule.name: "test"`), ensuring the category chosen matches the rule category
3. Create a rule that matches the scoped query and category
4. Generate some alerts
5. Assert the new alerts have the maintenance window ID associated 
6. Create a rule that doesn't match the scoped query, but matches the category
7. Generate some alerts
8. Assert the new alerts do not have the maintenance window ID associated
